cubedash/index/postgres/_api.py

```diff
@@ -422,40 +402,71 @@ def collections_search_query(
         time: tuple[datetime, datetime] | None = None,
         q: Sequence[str] | None = None,
     ) -> Result:
-        collection = self.collection_cols().alias("collection")
-        query = select(collection).where(collection.c.period_type == "all")
+        # STAC Collections only hold a bounding box in EPSG:4326, no geometry
+        # Calculate the bounding box on the server, it's far more efficient
+        collection_bbox = func.Box2D(
+            func.ST_Transform(TIME_OVERVIEW.c.footprint_geometry, 4326)
+        )
+        bbox_array = array(
+            [
+                func.ST_XMin(collection_bbox),
+                func.ST_YMin(collection_bbox),
+                func.ST_XMax(collection_bbox),
+                func.ST_YMax(collection_bbox),
+            ]
+        )
+        bbox_or_null = case(
+            (func.ST_XMin(collection_bbox).is_(None), None), else_=bbox_array
+        )
+        query = (
+            select(
+                ODC_PRODUCT.c.definition,
+                PRODUCT.c.name,
+                bbox_or_null.label("bbox"),
+                PRODUCT.c.time_earliest,
+                PRODUCT.c.time_latest,
+            )
+            .select_from(
+                TIME_OVERVIEW.join(
```

Comment:
I don't know anything about the table layout, are the joins deliberately inside the `select_from()` here but after it in the corresponding postgis code?

Reply:
They are deliberate, this is pretty much my first time writing SQLAlchemy code. I couldn't get it working with the exact same structure because the types of the Tables are different. In the legacy postgres section they're plain SQLAlchemy Tables. In the newer postgis section they're SQLAlchemy ORM Tables.

Another bug-bear with the postgis section is that it didn't go all the way in defining joins that SQLAlchemy can apply automatically, and it absolutely should. Doing that would probably make this query cleaner.